testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -915,22 +931,60 @@ impl BitcoinRPCRequest {
             jsonrpc: "2.0".to_string(),
         };
 
-        BitcoinRPCRequest::send(request_builder, payload)?;
+        BitcoinRPCRequest::send(&config, payload)?;
         Ok(())
     }
 
-    fn send(request_builder: RequestBuilder, payload: BitcoinRPCRequest) -> RPCResult<serde_json::Value> {
-        let body = json!(payload);
-        let result = request_builder.json(&body).send();
-        let response = result
-            .map_err(|e| RPCError::Network(format!("RPC Error: {}", e)))?;
+    fn send(config: &Config, payload: BitcoinRPCRequest) -> RPCResult<serde_json::Value> {
+        let mut request = BitcoinRPCRequest::build_rpc_request(&config);
+
+        let body = match serde_json::to_vec(&json!(payload)) {
+            Ok(body) => body,
+            Err(err) => {
+                return Err(RPCError::Network(format!("RPC Error: {}", err)));
+            }
+        };
+        request.append_header("Content-Type", "application/json").expect("Unable to set header");
+        request.append_header("Content-Length", format!("{}", body.len())).expect("Unable to set header");
+        request.set_body(body);
+
+        let mut response = async_std::task::block_on(async move {
```

Comment:
If we're just doing a synchronous request here, then why bother with `async-h1` and `async-std`?  Why not just use `reqwest`?

Reply:
reqwest contains unsafe code + pulls hyper, tokio, and 20 other dependencies

Reply:
Per the above, my concern is that we're now using _two_ HTTP client libraries (and neither of them will be included in the codebase when mainnet launches).

Reply:
One of them is a dev dependency, though. And I think should that these HTTP libraries (WARP and reqwest) should always remain a dev dependency -- the tests should not be rewritten to use our HTTP code, as using a convenient second HTTP implementation for tests helps insure that our HTTP code is compliant.

Reply:
I'm fine with using whatever we want for dev-dependencies.  I see now that `async-h1` does indeed replace all instances of `reqwest` in this PR for non-dev-dependencies.